runtime/compiler/optimizer/InlinerTempForJ9.cpp

```diff
@@ -3053,21 +3053,39 @@ bool TR_J9InlinerPolicy::tryToInlineTrivialMethod (TR_CallStack* callStack, TR_C
    TR_VirtualGuardSelection *guard = calltarget->_guard;
    TR::ResolvedMethodSymbol * callerSymbol = callStack->_methodSymbol;
 
-   if (isInlineableJNI(calleeSymbol->getResolvedMethod(),callNode))
+   if (!isInlineableJNI(calleeSymbol->getResolvedMethod(), callNode))
       {
-      if (performTransformation(comp(), "%sInlining jni %s into %s\n", OPT_DETAILS, calleeSymbol->signature(comp()->trMemory()), callerSymbol->signature(comp()->trMemory())))
-         {
-         if (calltarget->_myCallSite->isIndirectCall())
-            return true;
+      return false;
+      }
 
-         if (inlineGetClassAccessFlags(calleeSymbol, callerSymbol, callNodeTreeTop, callNode))
-            guard->_kind = TR_NoGuard;
-         else if (inlineUnsafeCall(calleeSymbol, callerSymbol, callNodeTreeTop, callNode))
-            guard->_kind = TR_NoGuard;
-         }
+   if (calltarget->_myCallSite->isIndirectCall())
+      {
+      return false;
+      }
+
+   if (!performTransformation(
+         comp(),
+         "%sInlining jni %s into %s\n",
+         OPT_DETAILS,
+         calleeSymbol->signature(comp()->trMemory()),
+         callerSymbol->signature(comp()->trMemory())))
+      {
+      return false;
+      }
+
+   if (inlineGetClassAccessFlags(calleeSymbol, callerSymbol, callNodeTreeTop, callNode))
```

Comment:
I did not check, but I assume these calls for inlining of class access flags and unsafe themselves cannot return without actually doing the transformation, e.g. due to some edge case. Even if they did, it was no different to the prior behavior, but I thought I would ask since the performTransformation has said that a transformation has taken place.

Reply:
I did take a quick look when I was writing this, and I believe they always return false on failure.

Reply:
The fact that the `performTransformation()` happens at a point where it isn't completely obvious that the transformation will succeed is also no different from the previous behaviour. That's why I added a new trace message in case of failure after that point, to prevent the log from being too misleading. I think that any target that `isInlineableJNI()` should be handled by one of these two functions (most likely `inlineUnsafeCall()`), but moving `performTransformation()` to the point where there is clearly a guarantee of success would require it to be pushed down into all of the various transformation functions, e.g. `createUnsafePutWithOffset()`, etc.